### PR TITLE
Add event submission/batching

### PR DESCRIPTION
**Pull request by nsalamad**
# Changes
* add EventQueue class
* change DVCOptions initializer to use named params
* make DVCClient properties private

### CODE REVIEW

File: lib/devcycle-ruby-server-sdk/event_queue.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    attr_accessor :logger
+    attr_accessor :max_event_queue_size
+    attr_accessor :sdk_key
+
+
+    sig { params(sdk_key: String, options: EventQueueOptions, local_bucketing: LocalBucketing).returns(NilClass) }
+    def initialize(sdk_key, options, local_bucketing)
+      @sdk_key = sdk_key
+      @events_api_uri = options.events_api_uri
+      @event_flush_interval_ms = options.event_flush_interval_ms || 10 * 1000
```
Comment (JamieSinn) [inline lib/devcycle-ruby-server-sdk/event_queue.rb:25]: Should we handle this defaulting in the DVC options as a getter instead?

File: lib/devcycle-ruby-server-sdk/event_queue.rb [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    def flush_events
+      payloads = @local_bucketing.flush_event_queue
+      if payloads.size == 0
+        return
+      end
+      eventCount = payloads.reduce(0) { |sum, payload| sum + payload.eventCount }
+      payloads.each do |payload|
+        begin
+          response = Typhoeus.post(
+            @events_api_uri + 'v1/events/batch',
```
Comment (JamieSinn) [inline lib/devcycle-ruby-server-sdk/event_queue.rb:71]: We need to include the starting `/` as the URI shouldn't include the trailing `/`

Comment (nsalamad) [reply lib/devcycle-ruby-server-sdk/event_queue.rb:71]: kk will fix

File: lib/devcycle-ruby-server-sdk/event_queue.rb (hunk 1 above)
Comment (nsalamad) [reply lib/devcycle-ruby-server-sdk/event_queue.rb:25]: yeah, that sounds good to me. I'll also move all the error checking there too

File: lib/devcycle-ruby-server-sdk/localbucketing/dvc_options.rb [hunk 3]
```diff
@@ -14,24 +14,23 @@ class DVCOptions
     attr_accessor :config_cdn_uri
     attr_accessor :events_api_uri
     attr_accessor :bucketing_api_uri
+    attr_accessor :event_request_chunk_size
```
Comment (JamieSinn) [inline lib/devcycle-ruby-server-sdk/localbucketing/dvc_options.rb:17]: it occurs to me that we shouldn't be setting these as accessors; as that way you can override and bypass our clamping after instantiation. Should probably be read only

Comment (nsalamad) [reply lib/devcycle-ruby-server-sdk/localbucketing/dvc_options.rb:17]: kk will make it read only

**Review by JamieSinn — APPROVED** [review]

**Pull request merged**